src/main/java/javax/validation/valueextraction/ExtractedValue.java

```diff
@@ -27,4 +28,20 @@
 @Documented
 public @interface ExtractedValue {
 
+	/**
+	 * Returns the type of the value extracted by the {@link ValueExtractor}. If not set, returns
+	 * {@code Void.class}, meaning the type will be automatically inferred from the type argument
+	 * of the parameterized type.
+	 * <p>
+	 * {@code type} is used to define {@code ValueExtractor}s for non parameterized type e.g.
```

Comment:
Minor style remark: I'd omit the name of the element ("{@code type}") as it's the one which is subject to this comment, i.e.:

> Used to define value extractors for non-generic wrapper types such as {@link OptionalInt}.